presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -690,7 +694,13 @@ public HiveInsertTableHandle beginInsert(ConnectorSession session, ConnectorTabl
 
         checkTableIsWritable(table.get());
 
-        HiveStorageFormat hiveStorageFormat = extractHiveStorageFormat(table.get());
+        HiveStorageFormat hiveStorageFormat;
+        if (respectTableFormat) {
+            hiveStorageFormat = extractHiveStorageFormat(table.get());
+        }
+        else {
+            hiveStorageFormat = getHiveStorageFormat(ImmutableMap.of());
```

Comment:
Oh, yeah, it's totally broken. It needs to create the hive storage format from the value in the config object.